test/test_basic.py

```diff
@@ -158,6 +175,31 @@ async def test_wrong_passphrase(self, key_dir: pathlib.Path, runtime_dir: pathli
     async def test_correct_passphrase(self, key_dir: pathlib.Path, runtime_dir: pathlib.Path) -> None:
         await self.run_test(key_dir, runtime_dir, True, 'passphrase')
 
+    @pytest.mark.asyncio
+    @pytest.mark.xfail
+    async def test_known_host_good(self, key_dir: pathlib.Path, runtime_dir: pathlib.Path) -> None:
+        await self.run_test(key_dir, runtime_dir, ZeroDivisionError(), 'passphrase',
+                            known_host_key='scan')
```

Comment:
This is really annoying. When doing this interactively, this works just fine:
```
❱❱❱ ssh -o UserKnownHostsFile=/tmp/pytest-of-martin/pytest-200/test_known_host_good0/keys/known_hosts -i /tmp/pytest-of-martin/pytest-200/test_known_host_good0/keys/id_rsa -p 39597 -l admin -o StrictHostKeyChecking=yes 127.0.0.1 hostname
cockpit-toolbox
```

but when I create an echo-only KnownHostsCommand:
```
❱❱❱ ssh -o UserKnownHostsFile=/tmp/pytest-of-martin/pytest-200/test_known_host_good0/keys/known_hosts -i /tmp/pytest-of-martin/pytest-200/test_known_host_good0/keys/id_rsa -p 39597 -l admin -o 'KnownHostsCommand=/tmp/xx %I %H %t %K %f' -o CheckHostIP=yes -o StrictHostKeyChecking=yes 127.0.0.1 hostname
YYY KnownHostsCommand ORDER 127.0.0.1 NONE NONE NONE
YYY KnownHostsCommand HOSTNAME [127.0.0.1]:39597 ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDWdxkgPs4niiaW41P8NiKjI3slCoeaRQvvchTHCyvQMGOanv+iudgurkcVvJOWOHsbLdxSfW5KbF1bGpVu3nwjbA7rajDx8Xs4z6VLsd4WCrHJl0qZt5GFfYTriIiPfE1t/C9MIxA1Vfxz099DBQDgs96kt7EidP2cBTb1rWGjBAt71jlfuxH4g1+emuPcuhdY3PFH5Ac7IwG5So3jxUWB7esDiO7StoKcAU2iJzp8yFLOrekYn8IA9cAxyzgYzlnqs8S/6aFrm/xTYAb2YIGyLUoyQgAIQW4MlILxq5opS5+YUYZaBLZRYoI3vkqqF+ULeqdZzgcOSLe4cbE3bZql SHA256:ZeBykpPEufJoNUuDWQTEnA+pIdkQUralwtLLItxK+GM
cockpit-toolbox
```

i.e. it still calls the KnownHostsCommand with a `HOSTNAME` **even if** it already knows it:

```
debug1: Found key in /tmp/pytest-of-martin/pytest-200/test_known_host_good0/keys/known_hosts:1
[...]
debug1: Will attempt key: /home/martin/.ssh/id_rsa RSA SHA256:Eq3Xz7MwpEieWi+5do3xJ9l0b/VW4ryM1B84W399zfg agent
debug1: Will attempt key: martin@cockpit-toolbox RSA SHA256:tIilj2EYf3gVPkTWIYURax+sLjyCSoodlsKZ4nRQVfk agent
debug1: Will attempt key: /tmp/pytest-of-martin/pytest-200/test_known_host_good0/keys/id_rsa RSA SHA256:VCdgVD78sNf5IiV49xlTPYqqrITt8o+uGUFEtJmKfOw explicit
```
So ssh's behaviour there is a bit misleading. I suppose one could interpret the behaviour that way:

> Specifies a command to use to obtain a list of host keys, in addition to those listed in UserKnownHostsFile and GlobalKnownHostsFile.  This command is executed after the files have been read.  It may write host key lines to standard output in identical format to the usual files (described in the VERIFYING HOST KEYS section in ssh(1)).

Reply:
See PR #16